File: dsa/Searching/Binarysearch.py
arr = [2, 4, 6, 8, 10, 12]
target = 8

low = 0
high = len(arr) - 1

while low <= high:
    mid = (low + high) // 2

    if arr[mid] == target:
        print("Element found at index", mid)
        break
    elif arr[mid] < target:
        low = mid + 1
    else:
        high = mid - 1
else:
    print("Element not found")

File: dsa/Searching/Binarysearch.java
public class BinarySearch {
    public static void main(String[] args) {

        int arr[] = {2, 4, 6, 8, 10, 12};
        int target = 8;

        int low = 0, high = arr.length - 1;

        while(low <= high) {

            int mid = (low + high) / 2;

            if(arr[mid] == target) {
                System.out.println("Element found at index " + mid);
                return;
            }
            else if(arr[mid] < target) {
                low = mid + 1;
            }
            else {
                high = mid - 1;
            }
        }

        System.out.println("Element not found");
    }
}